great_expectations/profiler/domain_builder/simple_semantic_type_domain_builder.py

```diff
@@ -61,12 +61,24 @@ def _get_domains(
                 self._get_column_semantic_type_name(validator=validator, column=column)
             )
             if semantic_column_type in semantic_types:
-                domains.append(
-                    Domain(
-                        domain_kwargs={"column": column},
-                        domain_type=semantic_column_type,
+                if include_batch_id:
+                    # TODO: <Alex>ALEX -- Should we use the "active_batch_id" or is there a reason to use one of the passed "batch_ids"?  Further, should we, in fact, include both, the "active_batch_id" as well as all "batch_ids" -- for multibatch case?</Alex>
```

Comment:
Just a thought - maybe to help users we add a `_get_all_domains()` to the base `DomainBuilder` and then call that and filter them here (since `_get_all_domains()` will likely be similar for all DomainBuilders)?
Also I'm not sure yet about including all batch_ids - thinking 🤔

Reply:
@anthonyburdi Per our brief analysis, it is not clear how to get *all* domains, since the building of domains follows the logic particular to the given domain builder.

Reply:
@alexsherstinsky very good point. In light of that instead of `DomainBuilder` I should have said that the domain-type specific builder e.g. `ColumnDomainBuilder` would get all domains. I'm sort of thinking of composing a domain builder from lists of two classes - one handles interacting with the appropriate domain type, and the other handles filtering. Something akin to solution 4 here: https://python-patterns.guide/gang-of-four/composition-over-inheritance/#solution-4-beyond-the-gang-of-four-patterns. I would love to hear your thoughts

Reply:
As per agreement, we are using `validator.active_batch_id` for now.